src/platform/nrfconnect/FactoryDataParser.c

```diff
@@ -84,6 +104,105 @@ static bool DecodeEntry(zcbor_state_t * states, void * buffer, size_t bufferSize
     return res;
 }
 
+bool SerializeFactoryData(uint8_t * factoryDataBuff, size_t factoryDataSize, const struct FactoryData * factoryData)
+{
+    zcbor_state_t cbor_state[MAX_FACTORY_DATA_NESTING_LEVEL];
+    zcbor_new_encode_state(cbor_state, MAX_FACTORY_DATA_NESTING_LEVEL, factoryDataBuff, factoryDataSize, 0);
+
+    if (!(zcbor_map_start_encode(cbor_state, 1) && zcbor_tstr_put_lit_cast(cbor_state, "version") &&
+          zcbor_uint32_put(cbor_state, (uint32_t) (factoryData->version)) && zcbor_tstr_put_lit_cast(cbor_state, "passcode") &&
```

Comment:
Can we keep C++ style for casting? (static_cast would fit)

Reply:
It's a C file.